sonoff/xdrv_16_tuyadimmer.ino

```diff
@@ -194,11 +195,11 @@ void TuyaPacketProcess(void)
     case TUYA_CMD_STATE:
       if (tuya_buffer[5] == 5) {  // on/off packet
 
-        AddLog_P2(LOG_LEVEL_DEBUG, PSTR("TYA: RX - %s State"),tuya_buffer[10]?"On":"Off");
-
-        if((power || Settings.light_dimmer > 0) && (power != tuya_buffer[10])) {
+        AddLog_P2(LOG_LEVEL_DEBUG, PSTR("TYA: RX Device -%d --> %s State"),tuya_buffer[6],tuya_buffer[10]?"On":"Off");
+        ExecuteCommandPower(tuya_buffer[6], tuya_buffer[10], SRC_SWITCH);  // send SRC_SWITCH? to use as flag to prevent loop from inbound states from faceplate interaction
```

Comment:
I think this may need to be handled differently.
Without the if it will likely break support for the "Touch Dimmer"